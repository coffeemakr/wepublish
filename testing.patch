diff --git a/libs/api/src/lib/db/article.ts b/libs/api/src/lib/db/article.ts
index 7359af6a..ea27be09 100644
--- a/libs/api/src/lib/db/article.ts
+++ b/libs/api/src/lib/db/article.ts
@@ -14,11 +14,11 @@ export interface ArticleData {
   readonly lead?: string | null
   readonly seoTitle?: string | null
   readonly slug?: string | null
-  readonly tags: string[]
+  readonly tags?: string[]
 
   readonly canonicalUrl?: string | null
 
-  readonly properties: MetadataProperty[]
+  readonly properties?: MetadataProperty[]
 
   readonly imageID?: string | null
   readonly authors: ArticleRevisionAuthor[]
@@ -107,7 +107,7 @@ export type ArticleRevisionWithRelations = PrismaArticleRevision & {
 
 export type ArticleWithRevisions = PrismaArticle & {
   draft: ArticleRevisionWithRelations | null
-  pending: ArticleRevisionWithRelations | null
+  pending?: ArticleRevisionWithRelations | null
   published: ArticleRevisionWithRelations | null
 }
 
diff --git a/libs/api/src/lib/db/page.ts b/libs/api/src/lib/db/page.ts
index fb73c289..6f0ed1b5 100644
--- a/libs/api/src/lib/db/page.ts
+++ b/libs/api/src/lib/db/page.ts
@@ -14,9 +14,9 @@ export interface PageData {
 
   readonly title: string
   readonly description?: string | null
-  readonly tags: string[]
+  readonly tags?: string[]
 
-  readonly properties: MetadataProperty[]
+  readonly properties?: MetadataProperty[]
 
   readonly imageID?: string | null
 
@@ -85,7 +85,7 @@ export type PageRevisionWithProperties = PrismaPageRevision & {
 
 export type PageWithRevisions = PrismaPage & {
   draft: PageRevisionWithProperties | null
-  pending: PageRevisionWithProperties | null
+  pending?: PageRevisionWithProperties | null
   published: PageRevisionWithProperties | null
 }
 
diff --git a/libs/api/src/lib/graphql/article.ts b/libs/api/src/lib/graphql/article.ts
index 8d80fb27..77357e01 100644
--- a/libs/api/src/lib/graphql/article.ts
+++ b/libs/api/src/lib/graphql/article.ts
@@ -275,7 +275,7 @@ export const GraphQLPublicArticle: GraphQLObjectType<PublicArticle, Context> =
         type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLMetadataPropertyPublic))),
         resolve: ({properties}) => {
           return properties
-            .filter(property => property.public)
+            ?.filter(property => property.public)
             .map(({key, value}) => ({key, value}))
         }
       },
diff --git a/libs/api/src/lib/graphql/article/article.private-mutation.ts b/libs/api/src/lib/graphql/article/article.private-mutation.ts
index 43a6a3ee..049c7e54 100644
--- a/libs/api/src/lib/graphql/article/article.private-mutation.ts
+++ b/libs/api/src/lib/graphql/article/article.private-mutation.ts
@@ -178,7 +178,8 @@ export const duplicateArticle = async (
       createMany: {
         data: socialMediaAuthors.map(({authorId}) => ({authorId}))
       }
-    }
+    },
+    blocks: []
   }
 
   return articleClient.create({
@@ -303,7 +304,8 @@ export const unpublishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -333,7 +335,8 @@ export const unpublishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           }
         }
       },
@@ -491,7 +494,8 @@ export const publishArticle = async (
                 createMany: {
                   data: socialMediaAuthors.map(({authorId}) => ({authorId}))
                 }
-              }
+              },
+              blocks: []
             },
             update: {
               ...revision,
@@ -521,7 +525,8 @@ export const publishArticle = async (
                 createMany: {
                   data: socialMediaAuthors.map(({authorId}) => ({authorId}))
                 }
-              }
+              },
+              blocks: []
             }
           }
         },
@@ -579,7 +584,8 @@ export const publishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -609,7 +615,8 @@ export const publishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           }
         }
       },
diff --git a/libs/api/src/lib/graphql/page.ts b/libs/api/src/lib/graphql/page.ts
index 56d380f0..6b63518e 100644
--- a/libs/api/src/lib/graphql/page.ts
+++ b/libs/api/src/lib/graphql/page.ts
@@ -210,7 +210,7 @@ export const GraphQLPublicPage = new GraphQLObjectType<PublicPage, Context>({
     properties: {
       type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLMetadataPropertyPublic))),
       resolve: ({properties}) => {
-        return properties.filter(property => property.public).map(({key, value}) => ({key, value}))
+        return properties?.filter(property => property.public).map(({key, value}) => ({key, value}))
       }
     },
 
diff --git a/libs/api/src/lib/graphql/page/page.private-mutation.ts b/libs/api/src/lib/graphql/page/page.private-mutation.ts
index 8d21a995..456df84e 100644
--- a/libs/api/src/lib/graphql/page/page.private-mutation.ts
+++ b/libs/api/src/lib/graphql/page/page.private-mutation.ts
@@ -140,7 +140,8 @@ export const duplicatePage = async (
       createMany: {
         data: duplicatedProperties
       }
-    }
+    },
+    blocks: []
   }
 
   return pageClient.create({
@@ -236,7 +237,8 @@ export const unpublishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -250,7 +252,8 @@ export const unpublishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           }
         }
       },
@@ -387,7 +390,8 @@ export const publishPage = async (
                 createMany: {
                   data: properties
                 }
-              }
+              },
+              blocks: []
             },
             update: {
               ...revision,
@@ -401,7 +405,8 @@ export const publishPage = async (
                 createMany: {
                   data: properties
                 }
-              }
+              },
+              blocks: []
             }
           }
         },
@@ -443,7 +448,8 @@ export const publishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -454,7 +460,8 @@ export const publishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           }
         }
       },
