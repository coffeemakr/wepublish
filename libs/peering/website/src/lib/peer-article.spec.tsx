import {runStorybookTests} from '@wepublish/testing'
import * as stories from './peer-information.stories'

describe('PeerInformation', () => {
  runStorybookTests(stories)
})
