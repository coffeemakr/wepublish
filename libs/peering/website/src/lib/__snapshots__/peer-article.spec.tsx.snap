// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeerInformation should render Default 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  padding: 24px;
  gap: 24px;
  justify-items: center;
  text-align: center;
  position: relative;
}

.emotion-2 {
  display: grid;
  grid-template-columns: 65px 1fr;
  gap: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 *:last-child {
  margin: 0;
}

@media (min-width:900px) {
  .emotion-6 {
    position: absolute;
    top: 24px;
    left: 24px;
  }
}

<aside
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    />
    <div
      class="emotion-4 emotion-5"
    />
    <div
      class="emotion-6 emotion-7"
    />
  </aside>
</DocumentFragment>
`;

exports[`PeerInformation should render WithClassName 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <DocumentFragment>
    <aside
-     class="css-1ng24i9-PeerInformationWrapper eyzn4eq3"
+     class="extra-classname css-1ng24i9-PeerInformationWrapper eyzn4eq3"
    >
      <div
        class="css-uc87i1-PeerInformationName eyzn4eq2"
      />
      <div"
`;

exports[`PeerInformation should render WithEmotion 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`PeerInformation should render WithError 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <DocumentFragment>
    <aside
      class="css-1ng24i9-PeerInformationWrapper eyzn4eq3"
-   >
-     <div
-       class="css-uc87i1-PeerInformationName eyzn4eq2"
-     />
-     <div
-       class="css-1eo3s8y-PeerInformationCTA eyzn4eq1"
    />
-     <div
-       class="css-198qeby-PeerInformationOrigin eyzn4eq0"
-     />
-   </aside>
  </DocumentFragment>"
`;

exports[`PeerInformation should render WithLoading 1`] = `
"Snapshot Diff:
- First value
+ Second value

- <DocumentFragment>
-   <aside
-     class="css-1ng24i9-PeerInformationWrapper eyzn4eq3"
-   >
-     <div
-       class="css-uc87i1-PeerInformationName eyzn4eq2"
-     />
-     <div
-       class="css-1eo3s8y-PeerInformationCTA eyzn4eq1"
-     />
-     <div
-       class="css-198qeby-PeerInformationOrigin eyzn4eq0"
-     />
-   </aside>
- </DocumentFragment>
+ <DocumentFragment />"
`;

exports[`PeerInformation should render WithoutCtaText 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`PeerInformation should render WithoutCtaUrl 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
